target/linux/ipq806x/image/Makefile

```diff
@@ -218,6 +237,29 @@ define Device/nec_wg2600hp3
 endef
 TARGET_DEVICES += nec_wg2600hp3
 
+define Device/MikrotikNAND
+	$(call Device/FitImageLzma)
+	DEVICE_VENDOR := MikroTik
+	KERNEL_INITRAMFS = $$(KERNEL) | ipq-aux-loader-kernel elf
+	KERNEL_INITRAMFS_SUFFIX := -fit-uImage.elf
+	PLATFORM := mikrotik-ipq806x
+	IMAGES := sysupgrade.bin
+	IMAGE/sysupgrade.bin := append-kernel | ipq-aux-loader-kernel ubi | sysupgrade-tar kernel=$$$$@ | append-metadata
+endef
+
+define Device/mikrotik_rb3011uias
+	$(call Device/MikrotikNAND)
+	DEVICE_MODEL := RB3011UiAS
+	SOC := qcom-ipq8064
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	AUX_LOADER_LEBSIZE := 126976
+	AUX_LOADER_LEBCOUNT := 120
+	AUX_LOADER_RB3011_PREINITS := YES
```

Comment:
If this is invented by you, please add it to DEVICEVARS

Reply:
This is part of the (removed) custom aux-loader. I'm still working on figuring out the alternative.